### PR TITLE
Fix missing external target settings with config conditions

### PR DESCRIPTION
**Pull request by fortmarek**
Resolves https://github.com/tuist/tuist/issues/6157

### Short description 📝

The issue with the Lookin dependency is that it has a `.define` setting with a config condition: https://github.com/QMUI/LookinServer/blob/develop/Package.swift#L53

While we decode that condition, we'd never actually use that setting.

The solution is to add the `.define` to the specified configuration settings.

### How to test the changes locally 🧐

- Generate `app_with_spm_dependencies`
- Ensure `SPM_LOOKIN_SERVER_ENABLED` is defined for Debug configuration
- Install the [lookin app](https://lookin.work/) and run the fixture app -> you should see the app in the Lookin app

### Contributor checklist ✅

- [x] The code has been linted using run `mise run lint:fix`
- [x] The change is tested via unit testing or acceptance testing, or both
- [x] The title of the PR is formulated in a way that is usable as a changelog entry
- [x] In case the PR introduces changes that affect users, the documentation has been updated

### Reviewer checklist ✅

- [x] The code architecture and patterns are consistent with the rest of the codebase
- [x] Reviewer has checked that, if needed, the PR includes the label `changelog:added`, `changelog:fixed`, or `changelog:changed`, and the title is usable as a changelog entry

### CODE REVIEW

File: Sources/TuistLoader/SwiftPackageManager/SettingsMapper.swift [hunk 1]
```diff
@@ -34,18 +34,43 @@ struct SettingsMapper {
...
+        return try map(
+            settings: platformSettings,
+            headerSearchPaths: headerSearchPaths,
+            defines: ["SWIFT_PACKAGE": "1"],
+            swiftDefines: "SWIFT_PACKAGE"
+        )
+    }
+
+    func settingsForPlatforms(_ platforms: [PackageInfo.Platform]) throws -> TuistGraph.SettingsDictionary {
+        var resolvedSettings = try settingsDictionaryForPlatform(nil)
```
Comment (pepicrft) [inline Sources/TuistLoader/SwiftPackageManager/SettingsMapper.swift:48]: What does it mean to get a settings dictionary for a platform `nil`? I'd iterate on the implementation to make the code more readable.

**Review by pepicrft — APPROVED** [review]

**Comment by topkim993** [comment]
I want to test it, but how can I install tuist and generate tuist locally?

Due to mise, it runs as 4.9.0.

`mise tuist@4.9.0 ✓ installed`

If there are any related documents, can I provide guidance?

**Comment by fortmarek** [comment]
@topkim993 The easiest way is to usually run tuist via Xcode as described [here](https://docs.tuist.io/contributors/get-started.html#run-tuist)

File: Sources/TuistLoader/SwiftPackageManager/SettingsMapper.swift (hunk 1 above)
Comment (fortmarek) [reply Sources/TuistLoader/SwiftPackageManager/SettingsMapper.swift:48]: If it's nil, it doesn't take a platform in the account. It was here before, but sure, let me refactor it 👍

**Comment by rakuyoMo** [comment]
@fortmarek @pepicrft Hello, Is there any new development here? I hope the next version will contain this fix, it's really important to us.

**Pull request merged**